subprojects/core/src/main/java/org/gradle/api/internal/artifacts/JavaEcosystemSupport.java

```diff
@@ -254,6 +261,25 @@ public void execute(CompatibilityCheckDetails<Bundling> details) {
         }
     }
 
+    @VisibleForTesting
+    public static class CategoryCompatibilityRules implements AttributeCompatibilityRule<Category>, ReusableAction {
+
+        @Override
+        public void execute(CompatibilityCheckDetails<Category> details) {
+            Category consumerValue = details.getConsumerValue();
+            if (consumerValue == null) {
+                // consumer didn't express any preferences, everything fits
+                details.compatible();
+                return;
+            }
+            String consumerValueName = consumerValue.getName();
+            if (Category.DOCUMENTATION.equals(consumerValueName)) {
+                // In the absence of documentation, everything else can act as 'documentation'
+                details.compatible();
```

Comment:
If I understand correctly, a consumer that says `org.gradle.category=documentation` will be able to consume all variants, `documentation` or not. But if the producer had two variants, one `documentation`, one `library`, now both will be considered. So you may need a disambiguation rule to make sure we prefer `documentation` when documentation is requested.